File: apps/uniapp-briefing/src/types/scheduled-report.ts
export type ReportStatus = 'ACTIVE' | 'PAUSED' | 'EXPIRED' | 'DELETED';
export type ScheduleType = 'cron' | 'interval' | 'daily' | 'weekly' | 'monthly' | 'once';
export type PushChannel = 'INAPP' | 'WECOM' | 'DINGTALK' | 'EMAIL' | 'UNI_PUSH' | 'SMS';

export interface ScheduledReport {
  report_id: number;
  tenant_id: number;
  user_id: number;
  name: string;
  description?: string;
  dsl_json: Record<string, any>;
  question?: string;
  render_blocks?: string[];
  chart_spec?: Record<string, any>;
  schedule_type: ScheduleType;
  cron_expr?: string;
  timezone?: string;
  next_run_at?: string;
  last_run_at?: string;
  fail_count?: number;
  status: ReportStatus;
  recipients?: string[];
  channels: PushChannel[];
  push_silent_if_empty?: boolean;
  push_format?: 'card' | 'table' | 'image' | 'excel';
  effective_from?: string;
  effective_to?: string;
  max_run_count?: number;
  created_at: string;
  updated_at: string;
  source?: 'chat' | 'manual' | 'template';
}

export interface ReportRun {
  run_id: number;
  report_id: number;
  scheduled_at: string;
  started_at?: string;
  finished_at?: string;
  duration_ms?: number;
  status: 'PENDING' | 'RUNNING' | 'SUCCESS' | 'FAILED' | 'EMPTY' | 'CANCELED';
  error_code?: string;
  error_msg?: string;
  rows_returned?: number;
  blocks_json?: any[];
  result_summary?: string;
  push_results?: Array<{ channel: string; status: string; latency_ms: number; error?: string }>;
  biz_tokens_charged?: number;
}

export interface ReportTemplate {
  template_id: number;
  business_line?: string;
  role?: string;
  category?: string;
  name: string;
  description?: string;
  dsl_json: Record<string, any>;
  cron_expr: string;
  channels: PushChannel[];
  popularity?: number;
}

export interface ScheduledReportInput {
  name: string;
  description?: string;
  dsl_json: Record<string, any>;
  question?: string;
  render_blocks?: string[];
  schedule_type: ScheduleType;
  cron_expr?: string;
  timezone?: string;
  recipients?: string[];
  channels: PushChannel[];
  push_silent_if_empty?: boolean;
  push_format?: string;
  effective_to?: string;
  max_run_count?: number;
  source?: string;
}

File: services/report-service/src/main/java/com/steel/report/model/ScheduledReport.java
package com.steel.report.model;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Map;

@Data
@NoArgsConstructor
public class ScheduledReport {
    private Long reportId;
    private Long tenantId;
    private Long userId;
    private String name;
    private String description;

    private Map<String, Object> dslJson;
    private String question;
    private List<String> renderBlocks;
    private Map<String, Object> chartSpec;

    private String scheduleType;
    private String cronExpr;
    private String timezone = "Asia/Shanghai";
    private LocalDateTime nextRunAt;
    private LocalDateTime lastRunAt;
    private Integer failCount = 0;
    private String status = "ACTIVE";

    private List<String> recipients;
    private List<String> channels;
    private Boolean pushSilentIfEmpty = false;
    private String pushFormat = "card";
    private String notifyTemplate;

    private LocalDate effectiveFrom;
    private LocalDate effectiveTo;
    private Integer maxRunCount;

    private Long costOwnerUserId;
    private Long estimatedBizTokens;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String source = "manual";
}
